### ISSUE TITLE
Bug in integration solver

### ISSUE THREAD

**Issue by pctablet505** [sympy/sympy#21721]
For calculating the integration of pdf of cauchy distribution I tried sympy, but it produced wrong result.

`integrate(1/(pi*(1+(x-a)**2)),(x,-oo,oo))` 
the **expected** output was **1** but it **produced output 0 which is wrong**.

![image](https://user-images.githubusercontent.com/55033230/124940822-ef6a4e80-e027-11eb-8772-214cbb31fd8a.png)

![image](https://user-images.githubusercontent.com/55033230/124941424-67387900-e028-11eb-833e-e1b44d074e06.png)
![image](https://user-images.githubusercontent.com/55033230/124941649-95b65400-e028-11eb-99cf-26861bdab0f3.png)

**Comment by oscarbenjamin** [comment]
If `a` is declared as real then the correct answer is given:
```python
In [8]: a = Symbol('a', real=True)

In [9]: integrate(1/(pi*(1+(x-a)**2)),(x,-oo,oo))
Out[9]: 1
```

**Comment by pctablet505** [comment]
Thank you so much :)

**Comment by oscarbenjamin** [comment]
Reopening because the answer should not be given as zero when `a` has not been declared as real

**Comment by oscarbenjamin** [comment]
Seems to be a bug in the limit code:
```python
In [1]: a = Symbol('a')

In [2]: I = integrate(1/(pi*(1+(x-a)**2)),x)

In [3]: I
Out[3]: 
  ⅈ⋅log(-a + x - ⅈ)   ⅈ⋅log(-a + x + ⅈ)
- ───────────────── + ─────────────────
          2                   2        
───────────────────────────────────────
                   π                   

In [4]: I.limit(x, oo)
Out[4]: 0

In [5]: I.limit(x, -oo)   # <-- should be -1
Out[5]: 0
```
Seems to be also a bug for this limit (even though the integral works in this case):
```python
In [9]: a = Symbol('a', real=True)

In [10]: I = integrate(1/(pi*(1+(x-a)**2)),x)

In [11]: I
Out[11]: 
-atan(a - x) 
─────────────
      π      

In [12]: I.limit(x, oo)  # <-- should be +1/2
Out[12]: -1/2

In [13]: I.limit(x, -oo)
Out[13]: -1/2
```

**Comment by oscarbenjamin** [comment]
> ```python
> In [12]: I.limit(x, oo)  # <-- should be +1/2
> Out[12]: -1/2
> ```

Actually this is a regression on master since sympy 1.8. Bisected to 2b6b23370113ac3460fba8e200a89416dfbb6cec from https://github.com/sympy/sympy/pull/21589
CC @0sidharth @jksuom

**Comment by jksuom** [comment]
```
In [4]: I.limit(x, oo)
Out[4]: 0

In [5]: I.limit(x, -oo)   # <-- should be -1
Out[5]: 0
```
I get these results also on sympy-1.8. It looks like that would be correct if `im(a) > 1` so that the arguments of the two log terms are on the same side of the real axis (on the same branch).

**Comment by oscarbenjamin** [comment]
> I get these results also on sympy-1.8.

I wonder if it depends on the Python version because I do not get the same in sympy 1.8 with Python 3.6 on Linux:
```python
(36venv) enojb@it051545:~/current/sympy/sympy.git$ git checkout sympy-1.8
HEAD is now at ed9a550... Merge pull request #21275 from oscarbenjamin/pr_release_script_ci
(36venv) enojb@it051545:~/current/sympy/sympy.git$ isympy
IPython console for SymPy 1.8 (Python 3.6.8-64-bit) (ground types: gmpy)

These commands were executed:
>>> from __future__ import division
>>> from sympy import *
>>> x, y, z, t = symbols('x y z t')
>>> k, m, n = symbols('k m n', integer=True)
>>> f, g, h = symbols('f g h', cls=Function)
>>> init_printing()

Documentation can be found at https://docs.sympy.org/1.8/


In [1]: a = Symbol('a', real=True)

In [2]: I = integrate(1/(pi*(1+(x-a)**2)),x)

In [3]: I
Out[3]: 
-atan(a - x) 
─────────────
      π      

In [4]: I.limit(x, -oo)
Out[4]: -1/2
```

**Comment by 0sidharth** [comment]
The limit regression has been fixed in #21731, although the answer is still returned as 0 when `a` is not declared real.

**Comment by oscarbenjamin** [comment]
Some of the problems mentioned above are fixed but some remain. The part that was identified as a regression has been fixed:
```python
In [28]: a = Symbol('a', real=True)

In [29]: I = integrate(1/(pi*(1+(x-a)**2)),x)

In [30]: I.limit(x, oo)  # <-- should be +1/2
Out[30]: 1/2
```
I'm removing the 1.9 milestone.
